include/ProgramRepresentation/FullFile.h

```diff
@@ -3,8 +3,26 @@
 
 #include "ProgramRepresentation/ProgramFunction.h"
 
-// reflects a file. this class manages all the functions that make up a single
-// file.
+/* reflects a file. this class manages all the functions that make up a single
+file. how everything in `ProgramRepresentation` is structured is as follows:
+
+- a `FullFile` is composed of functions (`ProgramFunction` s) that the user declares.
+- `ProgramFunction` s are composed of LLVM IR branches (`ProgramPoints` s)
+- and these branches contain information mapping each variable (and any of its aliases)
+to a set of methods called on them, which are the `DisjointPVAliasSets`.
+- these disjoint sets are composed of `PVAliasSet` s, which holds a set of program variables
+(typed `ProgramVariable`) which are must-aliases of each other, and the set of methods
+called on them (typed `MethodSet`)
+- `ProgramVariable` s are thus primitive objects that are isomorphic to the
+set of variables seen in the user's C code (or LLVM IR code – usually intermediate
```

Comment:
I think the text in parentheses is actually correct: `ProgramVariable` refers to an LLVM IR variable. Each of these is either a variable from the C code or an intermediate variable. Please rephrase this to be more clear.